hydra-cluster/src/Hydra/Cluster/Faucet.hs

```diff
@@ -88,6 +97,55 @@ findFaucetUTxO RunningNode{networkId, nodeSocket} lovelace = do
       FaucetHasNotEnoughFunds{faucetUTxO}
   pure foundUTxO
 
+seedFromFaucetBlockfrost ::
+  -- | Recipient of the funds
+  VerificationKey PaymentKey ->
+  -- | Amount to get from faucet
+  Coin ->
+  Blockfrost.BlockfrostClientT IO UTxO
+seedFromFaucetBlockfrost receivingVerificationKey lovelace = do
+  (faucetVk, faucetSk) <- liftIO $ keysFor Faucet
+
+  Blockfrost.Genesis
+    { Blockfrost._genesisNetworkMagic = networkMagic
+    , Blockfrost._genesisSystemStart = systemStart'
+    } <-
+    Blockfrost.queryGenesisParameters
+  pparams <- Blockfrost.queryProtocolParameters
+  let networkId = Blockfrost.toCardanoNetworkId networkMagic
+  let changeAddress = buildAddress faucetVk networkId
+  let receivingAddress = buildAddress receivingVerificationKey networkId
+  let theOutput =
+        TxOut
+          (shelleyAddressInEra shelleyBasedEra receivingAddress)
+          (lovelaceToValue lovelace)
+          TxOutDatumNone
+          ReferenceScriptNone
+  stakePools' <- Blockfrost.listPools
+  let stakePools = Set.fromList (Blockfrost.toCardanoPoolId <$> stakePools')
+  let systemStart = SystemStart $ posixSecondsToUTCTime systemStart'
+  eraHistory <- Blockfrost.queryEraHistory
+  foundUTxO <- findUTxO networkId changeAddress lovelace
+  case buildTransactionWithPParams' pparams systemStart eraHistory stakePools (mkVkAddress networkId faucetVk) foundUTxO [] [theOutput] of
+    Left e -> liftIO $ throwIO $ FaucetFailedToBuildTx{reason = e}
+    Right tx -> do
+      let signedTx = signTx faucetSk tx
+      eResult <- Blockfrost.tryError $ Blockfrost.submitTransaction signedTx
+      case eResult of
+        Left err -> liftIO $ throwIO $ FaucetBlockfrostError{blockFrostError = show err}
+        Right _ -> do
+          void $ Blockfrost.awaitUTxO networkId [changeAddress] (txId signedTx) 100
+          Blockfrost.awaitUTxO networkId [receivingAddress] (txId signedTx) 100
+ where
+  findUTxO networkId address lovelace' = do
+    faucetUTxO <- Blockfrost.queryUTxO networkId [address]
+    let foundUTxO = UTxO.find (\o -> (selectLovelace . txOutValue) o >= lovelace') faucetUTxO
+    when (isNothing foundUTxO) $
+      liftIO $
+        throwIO $
+          FaucetHasNotEnoughFunds{faucetUTxO}
+    pure $ maybe mempty (uncurry UTxO.singleton) foundUTxO
```

Comment:
what is the difference between mempty and having not founds?

Reply:
in this context it should be the same?